### ISSUE TITLE
zsh:51: command not found: zsh when trying to ssh

### ISSUE THREAD

**Issue by hermeschen1116** [warpdotdev/Warp#3880]
### Operating System

MacOS

### Operating System and Version

macOS 14.2 Beta (23C5047e)

### Local Shell Version

5.9.0.1-dev

### Remote Shell Version

5.9

### Warp Version

v0.2023.11.14.08.02.stable_02

### I have tried commenting out my system & user rcfiles and warp still will not bootstrap

- [X] Yes

### Screenshots

[comment output](https://app.warp.dev/block/f89ScboGnNAoQRkhoVPkLy)

### Include shell xtrace output

WARP_DEBUG_MODE=1 ssh Server
HermesChen@(my server)'s password:
zsh:51: command not found: zsh
Connection to (my server) closed.

### Does this block you from using Warp daily?

Yes, this issue prevents me from using Warp daily.

### Is this a Warp specific issue? (i.e. does it happen in Terminal, iTerm, Kitty, etc.)

Yes, this I confirmed this only happens in Warp, not other terminals.

### Warp Internal (ignore) - linear-label:e7dfaa84-5fdb-4a00-b754-d8912da923fa

None

### Warp Internal (ignore): linear-label:b8107fdf-ba31-488d-b103-d271c89cac3e

None

**Comment by dannyneira** [comment]
hey @hermeschen1116 Sorry to hear about this blocker. Please check on the remote machine and make sure that zsh is set as the default shell (check with `echo $SHELL`). If it is, you can also try the alternative method of connecting thru `/usr/bin/ssh`, make sure to add that to your Settings > Subshells menu, and it should give you the option to ["Warpify" the subshell](https://docs.warp.dev/features/subshells#how-to-warpify-the-subshell). 

To anyone else facing this issue, please add a :+1: to the original post at the top or comment with your details, and subscribe if you'd like to be notified.

**Comment by hermeschen1116** [comment]
Yeah, I'm sure default shell on all my device is zsh. I'll try warpify later but actually I face another problem in warp now. So maybe after I fix it.

**Comment by hermeschen1116** [comment]
> hey @hermeschen1116 Sorry to hear about this blocker. Please check on the remote machine and make sure that zsh is set as the default shell (check with `echo $SHELL`). If it is, you can also try the alternative method of connecting thru `/usr/bin/ssh`(see [ssh docs](https://docs.warp.dev/features/ssh#implementation)), make sure to add that to your Settings > Subshells menu, and it should give you the option to ["Warpify" the subshell](https://docs.warp.dev/features/subshells#how-to-warpify-the-subshell).
> 
> To anyone else facing this issue, please add a 👍 to the original post at the top or comment with your details, and subscribe if you'd like to be notified.

@dannyneira Hello, actually I don't know how to setup it. In my experience, when I trying to enter any ssh sessions, warp will open a subshell automatically without any configure in advance. But it doesn't happen recently. So I add ssh into the list in subshell setting, is this right?

**Comment by sinoon** [comment]
I encountered the same issue. The terminal provided by the MacOS system works fine, but I encountered problems when using warp.

**Comment by dannyneira** [comment]
@hermeschen1116 yes, put /usr/bin/ssh into your subshell menu, then run that instead of ssh ( e.g. `/usr/bin/ssh user@server`), then once connected it should popup the "Warpify" option.
